src/java/org/apache/cassandra/utils/Throwables.java

```diff
@@ -267,4 +270,30 @@ public static RuntimeException cleaned(Throwable t)
     {
         return unchecked(unwrapped(t));
     }
+
+    public static List<Throwable> causes(Throwable actual, boolean includeSuppressed)
+    {
+        if (actual.getCause() == null) return Collections.singletonList(actual);
+        List<Throwable> causes = new ArrayList<>();
+        forEach(actual, includeSuppressed, causes::add);
+        return causes;
+    }
+
+    public enum ForEachResult { ALL, PARTIAL }
+
+    public static ForEachResult forEach(Throwable actual, boolean includeSuppressed, Predicate<? super Throwable> fn)
```

Comment:
this is dead code I added... I was trying to do something else with `test/unit/org/apache/cassandra/utils/AssertionUtils.java` and went with a simpler solution; forgot to clean this up.

We should just drop, else we should dead with circular references